yarn-project/circuit-types/src/logs/l1_payload/l1_note_payload.ts

```diff
@@ -136,4 +147,26 @@ export class L1NotePayload extends L1Payload {
       this.noteTypeId.equals(other.noteTypeId)
     );
   }
+
+  static #getPublicValuesAndRemainingCipherText(input: Buffer): [Fr[], Buffer] {
+    const reader = BufferReader.asReader(input);
+    const numPubliclyDeliveredValues = reader.readUInt8();
+
+    const remainingData = reader.readToEnd();
+    const publicValuesData = remainingData.subarray(
+      remainingData.length - numPubliclyDeliveredValues * Fr.SIZE_IN_BYTES,
+      remainingData.length,
+    );
+    if (publicValuesData.length % Fr.SIZE_IN_BYTES !== 0) {
+      throw new Error(`Public values byte length is not a multiple of Fr size. Length: ${publicValuesData.length}`);
+    }
+    const publicValues = [];
+    for (let i = 0; i < publicValuesData.length; i += Fr.SIZE_IN_BYTES) {
+      publicValues.push(Fr.fromBuffer(publicValuesData.subarray(i, i + Fr.SIZE_IN_BYTES)));
+    }
+
+    const remainingCiphertext = remainingData.subarray(0, remainingData.length - publicValuesData.length);
+
+    return [publicValues, remainingCiphertext];
```

Comment:
Wouldn't it be easier to simply check the size of the buffer, subtract `numPubliclyDelivieredValues * Fr.SIZE_IN_BYTES`, read those many bytes into the ciphertext, and then read the rest into `publicValues`?

Reply:
Yes, that's much better. Addressed